test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -347,6 +350,36 @@ spec =
             fill 10 # no change
             atom.bar . should_equal 2
 
+        Test.specify "construct and fill with type" <|
+            pair = Meta.atom_with_hole (e -> My_Ascribed_Type.Value_With_Type 1 e 3)
+
+            atom = pair.value
+            fill = pair.fill
+
+            Meta.is_atom atom . should_be_true
+
+            atom.foo . should_equal 1
+            atom.baz . should_equal 3
+            case atom.bar of
+                n : Number -> Test.fail "Shouldn't be number yet: "+n
+                _ -> Nothing
```

Comment:
Personally IMO `atom.bar` here should be a dataflow error, i.e. a `case` on it should not execute at all.

I'd do:
```suggestion
            atom.bar . should_fail_with Some_Error_Type
```

Reply:
It is not dataflow error. The execution fails with:
```
Expected an error Type_Error but no error occurred, instead got: Meta.atom_with_hole
```
the value is a special `Meta.atom_with_hole` object currently.

Reply:
My point is - should it not be an error??

IMO yes, as otherwise we violate some rather sane assumptions.

If I have a value `b` created by constructor `Bar (x : Integer)`, then if I get a _value_ from `b.x`, I should be safe to assume that `b.x` is an `Integer`. Returning this temporary magical value will violate that assumption, whereas returning an error will not (at least in my intuitive model, an error is a divergent computation so it does not necessarily have a type).

Reply:
`Meta.atom_with_hole` is a powerful hack - it should be used with a care and the _hole should be filled_ as soon as possible. Typical use is implementing `map` with `List`. If the _hole escapes_ to the client code it is an error the one who's using `atom_with_hole`. That was the decision when _hole support_ was introduced.

Once I investigated returning [DataflowError](https://github.com/enso-org/enso/blob/7211c8317d60ab738ab00d460424a96057b4b31b/engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java#L27) as part of another issues, but I couldn't get that working easily. If you insist on using `DataflowError`, please report another issue. I can give it one more try.

Reply:
I see. Given it is an advanced feature not meant to be visible by end users, I guess it's not crucial, so if it's problematic I think we can leave it as is. We can revisit it if we ever get users tripping over this in practice.